### PR TITLE
Update Librewolf profile and Add Sway profile

### PR DESCRIPTION
**Pull request by vnepogodin**
Parent is shutting down then including firefox-common-addons.profile
[error.log](https://github.com/netblue30/firejail/files/6255116/error.log)

### CODE REVIEW

File: etc/profile-a-l/librewolf.profile [hunk 1]
```diff
@@ -19,6 +19,8 @@ whitelist ${HOME}/.librewolf
 #noblacklist ${HOME}/.mozilla
 #whitelist ${HOME}/.mozilla
 
+include whitelist-usr-share-common.inc
```
Comment (rusty-snake) [inline etc/profile-a-l/librewolf.profile:22]: That's firefox' block, should we add some of them too?
https://github.com/netblue30/firejail/blob/ba2d8a8e9bebd274fa5aa554e792164a7981e110/etc/profile-a-l/firefox.profile#L22-L28

File: etc/profile-a-l/sway.profile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Firejail profile for Sway
+# Description: i3-compatible Wayland compositor 
+# This file is overwritten after every install/update
+# Persistent local customizations
+include sway.local
+# Persistent global definitions
+include globals.local
+
+# all applications started in sway will run in this profile
+noblacklist ${HOME}/.config/sway
```
Comment (rusty-snake) [inline etc/profile-a-l/sway.profile:10]: Add a `blacklist` for it in disable-common.inc.

IIRC sway uses `~/.config/i3` as fallback if there is no `~/.config/sway`. We should add `nobalcklist ${HOME}/.config/i3`.

**Review by rusty-snake — COMMENTED** [review]
LGTM, some suggestions below.

Should we make sway redirect to i3?

**Comment by vnepogodin** [comment]
No it's should have own folder `$HOME/.config/sway`

And what about `$HOME/.config/i3`?
Sway doesn't work completely well with i3 config

File: etc/profile-a-l/librewolf.profile (hunk 1 above)
Comment (vnepogodin) [reply etc/profile-a-l/librewolf.profile:22]: no it's not needed

Comment (rusty-snake) [reply etc/profile-a-l/librewolf.profile:22]: It's not needed to start librewolf. But what is with other features. Does it support `/usr/share/webext`? Or do we want `/usr/share/doc`? (The questions are for anybody who has a opinion).

File: etc/profile-a-l/sway.profile (hunk 2 above)
Comment (vnepogodin) [reply etc/profile-a-l/sway.profile:10]: done

**Comment by BL4CKH47H4CK3R** [comment]
> LGTM, some suggestions below.
> 
> Should we make sway redirect to i3?

Sway is almost i3 the .config are also same the main diif is wayland and x11 but why need to redirect to i3. I think sway profile will also be cool. if not working then we can redirect to i3 for sure :+1:

**Comment by rusty-snake** [comment]
> Sway is almost i3 

That's the reason.

File: etc/profile-a-l/librewolf.profile (hunk 1 above)
Comment (glitsj16) [reply etc/profile-a-l/librewolf.profile:22]: I think we should add the relevant ones. AFAICT librewolf supports webext. So the only path NOT to whitelist would be /usr/share/firefox IMO.

**Comment by glitsj16** [comment]
@rusty-snake I agree it would make sense to implement sway as a redirect profile. Perhaps the term 'redirect' caused confusion in this context for other posters.

**Review by BL4CKH47H4CK3R — COMMENTED** [review]
@glitsj16, I know that desktop environment or windows manager doesn't matter when I am using firejail. But I see people are creating profile like for i3 which is redundant so I told. Even `archiso` is enough to run firejail :smile:

**Review by rusty-snake — DISMISSED** [review]
Re-add `include whitelist-usr-share-common.inc`.

**Comment by vnepogodin** [comment]
> Re-add `include whitelist-usr-share-common.inc`.

Why ?

**Comment by rusty-snake** [comment]
TL;DR: It whitelist lots of **important** `/usr/share` folders.
**Always use `include {allow,whitelist}-common.inc` in PRs.**

The role of `{allow,whitelist}-common.inc` is to make maintaining profiles easier and cause less breakage. *Always* use them in PRs unless you can 100% guarantee that they are not required. The paths to some resources can differ between distros, others depend on the HW where it runs or the configuration. Or the version you run.  Have you tested that it works on every major distro, with any configuration, any DE, any HW? It's not enough that it can start there. Can it play audio? Is HTTPS working? What about spellchecking? Just the first line  (`whitelist /usr/share/alsa`) can you guarantee that not `whitelist`ing `/usr/share/alase`  does not break sound? With raw alsa?

**Comment by vnepogodin** [comment]
> TL;DR: It whitelist lots of **important** `/usr/share` folders.
> 
> **Always use `include {allow,whitelist}-common.inc` in PRs.**
> 
> 
> 
> The role of `{allow,whitelist}-common.inc` is to make maintaining profiles easier and cause less breakage. *Always* use them in PRs unless you can 100% guarantee that they are not required. The paths to some resources can differ between distros, others depend on the HW where it runs or the configuration. Or the version you run.  Have you tested that it works on every major distro, with any configuration, any DE, any HW? It's not enough that it can start there. Can it play audio? Is HTTPS working? What about spellchecking? Just the first line  (`whitelist /usr/share/alsa`) can you guarantee that not `whitelist`ing `/usr/share/alase`  does not break sound? With raw alsa?

I agree but how about ?

https://github.com/netblue30/firejail/pull/4164#discussion_r606858628

**Comment by rusty-snake** [comment]
(Privacy is out of the scope of firejail, it's for security).

> its not needed as it reveals lots of important /usr/share folders

It's not needed because it reveals important folders???!

> like /usr/share/fonts which can used for font fingerprinting 

This does not prevent font fingerprinting. Actually it makes the situation worse. If you `whitelist /usr/share/fonts` and don't install to much other fonts you have the same font FP as any other user of your distro. If you omit it and put on a gas mask you're even more unique. 

> and OS detection

Your OS can be detected anyway.
- Linux has an other TCP/IP-Stack then Window => Passive detection possible.
- Some math operations on floating point number has other results on Linux then Window. => Active detection via JavaScript
- Your scrollbar-width depends on your GTK-Theme and version => Active detection via JS or CSS possible.
- Your HTTP-UA contains your OS.
- Your JS-UA contains your OS, even with RFP.
- ...

> Linux and windows common font are not same so its a problem.

Masking your fonts still does not make them the same.

> Librewolf can launch

yes

> work perfectly without this options

no (I'm 90% sure it can not open HTTPS on some system without wusc)

**Comment by rusty-snake** [comment]
BTW I don't think `whitelist /usr/share/gnome-shell/search-providers/firefox-search-provider.ini` (*firefox*-search-provider) is used.

File: etc/profile-a-l/librewolf.profile [hunk 3]
```diff
@@ -19,16 +19,40 @@ whitelist ${HOME}/.librewolf
...
 whitelist /usr/share/doc
-whitelist /usr/share/gnome-shell/search-providers/firefox-search-provider.ini
 whitelist /usr/share/gtk-doc/html
 whitelist /usr/share/mozilla
 whitelist /usr/share/webext
 
-# librewolf requires a shell to launch on Arch. We can possibly remove sh though.
-#private-bin bash,dbus-launch,dbus-send,env,librewolf,python*,sh,which
+# librewolf requires a shell to launch on Arch.
+#private-bin bash,dbus-launch,dbus-send,env,librewolf,sh,which
```
Comment (rusty-snake) [inline etc/profile-a-l/librewolf.profile:34]: Based on https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=librewolf#n325 it's just `private-bin librewolf,sh` if anyone want to test ...

File: etc/profile-a-l/librewolf.profile [hunk 4]
```diff
@@ -19,16 +19,40 @@ whitelist ${HOME}/.librewolf
...
+# Fedora use shell scripts to launch librewolf, at least this is required
+#private-bin basename,bash,cat,dirname,expr,false,librewolf,getenforce,ln,mkdir,pidof,restorecon,rm,rmdir,sed,sh,tclsh,true,uname
```
Comment (rusty-snake) [inline etc/profile-a-l/librewolf.profile:36]: I don't see a Fedora package of librewolf (https://librewolf-community.gitlab.io/install/) ... AppImage and Flatpak seem to be the only install option for (non hacker) fedora users ... remove this.

**Comment by rusty-snake** [comment]
This PR has a a merge conflict, please rebase.

File: etc/profile-a-l/librewolf.profile [hunk 5]
```diff
@@ -14,15 +14,45 @@ mkdir ${HOME}/.librewolf
...
+
+whitelist /usr/share/doc
+whitelist /usr/share/gtk-doc/html
+whitelist /usr/share/mozilla
+whitelist /usr/share/webext
+include whitelist-usr-share-common.inc
+
 # librewolf requires a shell to launch on Arch. We can possibly remove sh though.
+# Add the next line to your librewolf.local to enable private-bin.
 #private-bin bash,dbus-launch,dbus-send,env,librewolf,python*,sh,which
```
Comment (rusty-snake) [inline etc/profile-a-l/librewolf.profile:34]:
```suggestion
#private-bin bash,dbus-launch,dbus-send,env,librewolf,sh,which
```

**Review by rusty-snake — APPROVED** [review]

File: etc/profile-a-l/librewolf.profile (hunk 5 above)
Comment (vnepogodin) [reply etc/profile-a-l/librewolf.profile:34]: I've had added before but it can't merge changes

Comment (rusty-snake) [reply etc/profile-a-l/librewolf.profile:34]: After a rebase (unlike a merge of master) it should work w/o conflicting. Rebasing and squashing some of the commits makes sense anyway if you look at the history. So if you can do an interactive rebase.

Comment (vnepogodin) [reply etc/profile-a-l/librewolf.profile:34]: done :smiley:

File: etc/profile-a-l/librewolf.profile [hunk 6]
```diff
@@ -18,12 +18,41 @@ whitelist ${HOME}/.librewolf
...
+#whitelist ${RUNUSER}/org.keepassxc.KeePassXC.BrowserServer
+
+whitelist /usr/share/doc
+whitelist /usr/share/gtk-doc/html
+whitelist /usr/share/mozilla
+whitelist /usr/share/webext
+include whitelist-usr-share-common.inc
+
 # librewolf requires a shell to launch on Arch. We can possibly remove sh though.
 # Add the next line to your librewolf.local to enable private-bin.
```
Comment (rusty-snake) [inline etc/profile-a-l/librewolf.profile:33]: Maybe something like this (feel free to improve)?
```suggestion
# Add the next line to your librewolf.local to enable private-bin (Arch Linux).
```

BTW: For what does it need `dbus-launch`?

**Review by glitsj16 — APPROVED** [review]

**Review by rusty-snake — APPROVED** [review]

**Comment by rusty-snake** [comment]
merged, thanks!
